targets/targets.json

```diff
@@ -278,6 +278,12 @@
             "bare-metal"
         ],
         "device_name": "LPC1114FN28/102",
+        "config": {
```

Comment:
> MBED CLI1 expects that if any of the mem configs are present in targets.json should have bootloader support

This seems odd to me. We also have a `"bootloader_supported"` flag in `targets.json`, which we can set to `false`. Is `"bootloader_supported": false` not respected by mbed-cli1?

Reply:
If I am not wrong, Mbed CLI1 expected "bootloader_supported": true and raised an exception for false https://github.com/ARMmbed/mbed-os/blob/master/tools/config/__init__.py#L821

Reply:
Yeah, you're right according to that code snippet. Raising an exception there seems wrong to me. It would seem plausibly more appropriate to check `if bootloader_supported and mem_regions not defined` before raising, instead of checking `if mem_regions defined and not bootloader_supported`. Maybe I'm missing some context about why it is this way. Anyway, we probably can't fix it due to the tools being frozen.